### PR TITLE
[TSD-145] Remove unused modules and features

### PR DESCRIPTION
**Pull request by HirotoShioi**
YT issue: https://iohk.myjetbrains.com/youtrack/issue/TSD-145

I'm also putting some effort on Haddock comments. (It failed to build Haddock).

### CODE REVIEW

File: src/DataSource/Types.hs [hunk 1]
```diff
@@ -98,14 +98,14 @@ runApp (App a) = runReaderT a
 
 -- | The basic configuration.
 data Config = Config
-    { cfgAgentId            :: !Integer
-    -- ^ Zendesk agent id
+    { cfgAgentId            :: !UserId
```
Comment (ksaric) [inline src/DataSource/Types.hs:102]: Nice.

**Review by ksaric — APPROVED** [review]
LGTM.

**Review by ksaric — CHANGES_REQUESTED** [review]
Fix test.

**Comment by HirotoShioi** [comment]
@ksaric Will do. Thank you!

**Comment by HirotoShioi** [comment]
I've finished adding Haddock comments. The project now has decent coverage.

**Review by ksaric — APPROVED** [review]
Very nice work!

**Pull request merged**